From a59b808905fc3017d1d8d137842946dd183d8ba2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:26:28 +0800
Subject: [PATCH] Revert "USB: gadget: composite: Don't increment interface
 number for alt settings."

This reverts commit b1d4000fe60cec6a05fd09b7a6697ede7ab649a2.
---
 drivers/usb/gadget/composite.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 08bcecbb934c..aac724ae664b 100755
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -322,13 +322,8 @@ static int config_buf(struct usb_configuration *config,
 		dest = next;
 		while ((descriptor = *descriptors++) != NULL) {
 			intf = (struct usb_interface_descriptor *)dest;
-			if (intf->bDescriptorType == USB_DT_INTERFACE) {
-				/* don't increment bInterfaceNumber for alternate settings */
-				if (intf->bAlternateSetting == 0)
-					intf->bInterfaceNumber = interfaceCount++;
-				else
-					intf->bInterfaceNumber = interfaceCount - 1;
-			}
+			if (intf->bDescriptorType == USB_DT_INTERFACE)
+				intf->bInterfaceNumber = interfaceCount++;
 			dest += intf->bLength;
 		}
 
-- 
2.35.3

